src/ORM/Search/SearchContext.php

```diff
@@ -176,40 +205,117 @@ public function getQuery($searchParams, $sort = false, $limit = false, $existing
+                        // where ((field1 like lorem OR field2 like lorem) AND (field1 like ipsum OR field2 like ipsum))
+                        $primarySubGroup = $dataQuery->conjunctiveGroup();
+                        foreach (explode(' ', $phrase) as $searchTerm) {
+                            $searchAcrossFields($searchTerm, $primarySubGroup->disjunctiveGroup());
+                        }
+                    }
+                } else {
+                    // where ((field1 like lorem OR field2 like lorem) AND (field1 like ipsum OR field2 like ipsum))
+                    $primarySubGroup = $dataQuery->conjunctiveGroup();
+                    foreach (explode(' ', $searchPhrase) as $searchTerm) {
+                        $searchAcrossFields($searchTerm, $primarySubGroup->disjunctiveGroup());
                     }
                 }
+            } else {
+                // where (field1 like lorem ipsum OR field2 like lorem ipsum)
+                $searchAcrossFields($searchPhrase, $dataQuery->disjunctiveGroup());
             }
+        });
+    }
+
+    /**
+     * Get the search filter for the given fieldname when searched from the primary search field.
+     */
+    public function getPrimarySearchFilter(string $modelClass, string $fieldName): ?SearchFilter
+    {
+        if ($filterClass = Config::inst()->get($modelClass, 'primary_search_field_filter')) {
+            return Injector::inst()->create($filterClass, $fieldName);
         }
+        return $this->getFilter($fieldName);
+    }
 
+    /**
+     * Search against a single field
+     *
+     * @param string|array $searchPhrase
+     */
+    protected function individualFieldSearch(DataList $query, array $searchableFields, string $searchField, $searchPhrase): DataList
+    {
+        if ($filter = $this->getFilter($searchField)) {
+            $filter->setModel($this->modelClass);
+            $filter->setValue($searchPhrase);
+            $searchableFieldSpec = $searchableFields[$searchField] ?? [];
+            return $query->alterDataQuery(function ($dataQuery) use ($filter, $searchableFieldSpec) {
+                $this->applyFilter($filter, $dataQuery, $searchableFieldSpec);
+            });
+        }
         return $query;
     }
 
+    /**
+     * Apply a SearchFilter to a DataQuery for a given field's specifications
+     */
+    public function applyFilter(SearchFilter $filter, DataQuery $dataQuery, array $searchableFieldSpec): void
+    {
+        if (!$filter->isEmpty()) {
+            if (isset($searchableFieldSpec['match_any'])) {
+                $searchFields = $searchableFieldSpec['match_any'];
+                $filterClass = get_class($filter);
+                $value = $filter->getValue();
+                $modifiers = $filter->getModifiers();
+                $subGroup = $dataQuery->disjunctiveGroup();
+                foreach ($searchFields as $matchField) {
+                    /** @var SearchFilter $filterClass */
```

Comment:
Put this docblock type directly above the $filterClass assignment

Reply:
Should actually be for `$filter` - will update the variable name in the docblock instead.

Reply:
done